lib/engine/game/g_1871/step/dividend.rb

```diff
@@ -31,8 +31,17 @@ def share_price_change(_entity, revenue)
           end
 
           def payout_per_share(entity, revenue)
+            share_count =
+              if entity == @game.peir
+                # entity.total_shares doesn't always work correctly for PEIR.
+                # If there are three shares then each is a 33% share, and
+                # total_shares returns 3.030303...
```

Comment:
imho, the comment isn't necessary. If someone is curious, they can look at the commit that introduced it and know why.